src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -11672,6 +11672,8 @@ bool Compiler::fgExpandRarelyRunBlocks()
 
                 // Check for a BBJ_CALLFINALLY followed by a rarely run paired BBJ_ALWAYS
                 //
+                // TODO-Cleanup: How can this be hit? If bbPrev starts a CallAlwaysPair, then this
+                // block must be BBJ_ALWAYS, not BBJ_CALLFINALLY.
```

Comment:
Probably should have been checking `block->isBBCallAlwaysPair`.

Likely we almost never see this case, it must be from something like an IBC compile where the finally is may throw and always throws in the IBC scenarios, so that the first block in the pair has nonzero weight, and the second block zero.